### PR TITLE
feat: Support sorting on primary key fields

### PR DESCRIPTION
**Pull request by himank**
## Describe your changes
- Refactoring query planner
- Honoring the sorting if it is on primary key fields
- Strictly enforcing sorting on a single field for database queries.

### CODE REVIEW

File: query/filter/key_builder.go [hunk 1]
```diff
@@ -422,11 +442,8 @@ func QueryPlanFromSort(sortFields *[]tsort.SortField, indexeableFields []*schema
 		return nil, nil
 	}
 
-	if len(*sortFields) > 1 {
-		return nil, errors.InvalidArgument("Cannot query with multiple sort fields")
-	}
 	var field *schema.QueryableField
-	for _, idx := range indexeableFields {
+	for _, idx := range indexableFields {
```
Comment (garrensmith) [inline query/filter/key_builder.go:446]: Thanks for the spelling fix.

File: server/services/v1/database/query_planner.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"github.com/tigrisdata/tigris/keys"
+	"github.com/tigrisdata/tigris/query/filter"
+	"github.com/tigrisdata/tigris/query/sort"
+	"github.com/tigrisdata/tigris/schema"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/value"
+)
+
+type PrimaryIndexPlanner struct {
```
Comment (garrensmith) [inline server/services/v1/database/query_planner.go:26]: This is very nice

**Review by garrensmith — CHANGES_REQUESTED** [review]
Some small changes but this is looking good.

File: query/filter/key_builder.go [hunk 3]
```diff
@@ -157,7 +173,7 @@ func (k *KeyBuilder[F]) Build(filters []Filter, userDefinedKeys []F) ([]QueryPla
 	for len(queue) > 0 {
 		element := queue[0]
 		if e, ok := element.(LogicalFilter); ok {
-			if e.Type() == OrOP && !k.primaryKey {
+			if e.Type() == OrOP && k.indexType == SecondaryIndex {
```
[the remark this answers is not in the document]
Comment (himank) [reply query/filter/key_builder.go:176]: Sure

File: server/services/v1/database/base_runner.go [hunk 4]
```diff
@@ -252,11 +232,14 @@ func (runner *BaseQueryRunner) mustBeDocumentsCollection(collection *schema.Defa
...
+func (runner *BaseQueryRunner) getSortOrdering(coll *schema.DefaultCollection, sortReq jsoniter.RawMessage, forSearch bool) (*sort.Ordering, error) {
+	ordering, err := sort.UnmarshalSort(sortReq)
 	if err != nil || ordering == nil {
 		return nil, err
 	}
+	if !forSearch && len(*ordering) > 1 {
+		return nil, errors.InvalidArgument("Cannot query with multiple sort fields")
+	}
 
 	for i, sf := range *ordering {
```
[the remark this answers is not in the document]
Comment (himank) [reply server/services/v1/database/base_runner.go:244]: That's right, `InMemoryName` is our alias in search only to avoid conflicts.

**Review by efirs — APPROVED** [review]

**Pull request merged**